lms/static/js/ccx/schedule.js

```diff
@@ -104,23 +104,37 @@ var edx = edx || {};
 	    // Add unit handler
 	    $('#add-unit-button').on('click', function(event) {
 	      event.preventDefault();
+	      // Default value of time is 00:00.
+	      var start = self.get_datetime('start');
 	      var chapter = self.chapter_select.val(),
-		  sequential = self.sequential_select.val(),
-		  vertical = self.vertical_select.val(),
-		  units = self.find_lineage(self.schedule,
-		    chapter,
-		    sequential === 'all' ? null : sequential,
-		    vertical === 'all' ? null: vertical),
-		  start = self.get_datetime('start'),
-		  due = self.get_datetime('due');
-	      units.map(self.show);
-	      var unit = units[units.length - 1];
-	      if (unit !== undefined && start) { unit.start = start; }
-	      if (unit !== undefined && due) { unit.due = due; }
-	      self.schedule_apply([unit], self.show);
-	      self.schedule_collection.set(self.schedule);
-	      self.dirty = true;
-	      self.render();
+	      sequential = self.sequential_select.val(),
+	      vertical = self.vertical_select.val(),
+	      units = self.find_lineage(
+	        self.schedule,
+	        chapter,
+	        sequential === 'all' ? null : sequential,
+	        vertical === 'all' ? null : vertical
+	      ),
+	      due = self.get_datetime('due');
+	      var errorMessage = self.valid_dates(start, due);
+	      if (errorMessage === undefined) {
+	        units.map(self.show);
+	        var unit = units[units.length - 1];
+	        if (unit !== undefined && start) {
+	          unit.start = start;
+	        }
+	        if (unit !== undefined && due) {
+	          unit.due = due;
+	        }
+	        self.schedule_apply([unit], self.show);
+	        self.schedule_collection.set(self.schedule);
+	        self.dirty = true;
+	        self.render();
+				} else {
+	        $('#ccx_schedule_error_message').html(gettext(errorMessage));
```

Comment:
Shouldn't you be calling [`.text()`](http://api.jquery.com/text/) instead of `.html()`? The way this code is written now, it looks like a potential opening for an XSS attack.

Also, you're calling `gettext()` on a variable instead of on a string literal, which is a code smell. Where is this `errorMessage` coming from, and why isn't it getting internationalized there? If it *is* getting internationalized there, you shouldn't call `gettext()` on it a second time.